### PR TITLE
Feat/team statistics

### PR DESCRIPTION
**Pull request by bwbolt**
### What does this PR do?
- This PR adds all base tests of the team statistics methods
- It also adds all the methods to make those tests pass

### Does this PR break anything?
- [ ] Yes
- [X] no
- If yes explain:

### CODE REVIEW

**Comment by bwbolt** [comment]
Its definitely not pretty but the old "red,green,refactor" is what im going after

**Review by z-prince — APPROVED** [review]
Great work creating those extra methods to support the required ones.

File: spec/team_statistics_spec.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  end
+
+  it 'can produce the team that wins against a team the least' do
+    expect(@stat_tracker.favorite_opponent('18')).to eq 'DC United'
+  end
+
+  it 'can produce the team that wins against a team the most' do
+    expect(@stat_tracker.rival('18')).to eq('Houston Dash').or(eq('LA Galaxy'))
+  end
+end
```
Comment (pdthomson) [inline spec/team_statistics_spec.rb:64]: everything looks great, APPROVED

**Review by pdthomson — APPROVED** [review]
Code looks great for both the tests and the methods I see no issues. APPROVED

**Pull request merged**